packages/fluentui/react-builder/src/narration/DescendantsNarrationsComputer.tsx

```diff
@@ -0,0 +1,80 @@
+import { NarrationComputer, IAriaElement } from './NarrationComputer';
+
+interface INarration {
+  path: string[];
+  text: string;
+}
+
+export class DescendantsNarrationsComputer {
+  computer: NarrationComputer = new NarrationComputer();
+
+  // Computes the screen reader narrations for the given element and all its focusable descendants, for the given platform.
+  async compute(element: IAriaElement, platform: string): Promise<INarration[]> {
+    // Prepare all the arrays
+    const activeDescendantsParents: IAriaElement[] = [];
+    const path: string[] = [];
+    const narrations: INarration[] = [];
+
+    // Traverse down the DOM tree rooted at the element and compute the activeDescendantsParents, path and narrations along the way
+    this.findActiveDescendantsParents(element, activeDescendantsParents);
+    await this.traverse(element, platform, path, narrations, activeDescendantsParents);
+
+    return narrations;
+  } // End compute
+
+  // Traverses the DOM tree rooted at the given element to find all the parents which have the ariaActiveDescendantElement property set, and saves the found elements to the given parents array.
+  findActiveDescendantsParents(element: IAriaElement, parents: IAriaElement[]) {
+    if (element.ariaActiveDescendantElement != null) {
+      // Begin if 1
+      parents.push(element);
+    } // End if 1
+    for (let i = 0; i < element.children.length; i++) {
+      // Begin for 1
+      const child = element.children[i] as IAriaElement;
+      this.findActiveDescendantsParents(child, parents);
+    } // End for 1
```

Comment:
nit: might be cleaner ? 

```suggestion
    element.children.forEach((child: IAriaElement) => this.findActiveDescendantsParents(child, parents))
```

Reply:
That won't work because `HTMLCollection` is not an Array object but an array-like object which doesn't have the `forEach()` method, so I get "TS2339: Property 'forEach' does not exist on type 'HTMLCollection'." Neither it can be iterated using `for/of` loop, in which case I get "TS2495: Type 'HTMLCollection' is not an array type or a string type." See more explanation [here](https://stackoverflow.com/questions/53098742/typescript-for-loop-over-htmlcollection). However, I  can use `Array.from()` to transform the `HTMLCollection` into the Array object, and then use `forEach()`.

Reply:
Your decision. Some people might scream that `Array.from` could iterate the array a second time, but if performance is not an issue, I consider my suggestion to be more easily readable (this is subjective though)

Reply:
Performance is not an issue here, so I will leave it using `forEach()`. If nothing, I at least learned something new :-)